docs/source/miscellaneous/doc/adr/0014-cassandra-targeted-sweep.md

```diff
@@ -0,0 +1,283 @@
+  be lower than 100 MB. We allow a maximum of 100_000 entries per dedicated row, which ensures that each dedicated row should not be
+  larger than 300MB while still allowing up to 6.4 million writes in a single transaction, and in practice even more, as long as
+  the number of shards is greater than 1. Note that for each cell in a non-dedicated row, we can calculate the write's start
+  transaction simply by multiplying the *timestamp_partition* by *50_000* and then adding its *timestamp_modulus*.
+
+#### sweepableTimestamps
+
+  This is an auxiliary table for locating the next row of the *sweepableCells* table to read since the timestamp partitions can be
+  sparse, and therefore requiring many lookups to locate a nonempty row. Each non-dedicated row of *sweepableCells* is represented
+  by a single cell in this table.
+
+  **Row components**:
+  - **shard**: a `VAR_LONG` containing the shard for which the row has entries for.
+  - **timestamp_partition**: a `VAR_LONG` corresponding to the *coarse timestamp partition* of all entries in the row.
+  - **sweep_conservative**: a `boolean` (encoded as a `BLOB`) specifying if the row contains entries for thorough (`0x00`) or
+    conservative sweep (`0x01`).
+
+  **Column components**:
+  - **timestamp_modulus** is the **fine timestamp partition** of a row in SweepableCells that falls into the coarse partition
+    specified in the timestamp_partition row component.
+
+  **Value**: unused empty byte array.
+
+  Locating the next row of *sweepableCells* to read is then done by reading the next column in the coarse partition we are currently
+  in, or range scanning the much denser *sweepableTimestamps* table to locate the next coarse partition with entries and then reading
+  the fine partition to look for.
+
+#### sweepProgressPerShard
+
+  This table stores targeted sweep's progress, as well as the information about the number of shards the sweep queue is using.
+
+  **Row components**:
+  - **shard**: a `VAR_SIGNED_LONG` containing the shard for which we are tracking progress for.
+  - **sweep_conservative**: a `boolean` (encoded as a `BLOB`) specifying if we are looking for conservative or thorough sweep as in
+    *sweepableTimestamps*.
+
+  **Named Column**:
+  - **value**: a `VAR_LONG` containing the timestamp up to which targeted sweep has swept on that shard and strategy.
+
+  To persist the number of shards sweep queue is using, we use a distinguished row of this table, with the row defined by
+  *shard = -1* and *sweep_conservative = true*. Note that all writes to this table use atomic check and set, and the values are only
+  allowed to be increased. A request to update a cell to a lower value will have no effect.
+
+#### Writing to the Sweep Queue
+
+Whenever a transaction is about to commit, before its writes are persisted into the KVS, `SerializableTransactionManager` enqueues
```

Comment:
technically `SnapshotTransaction` has the code to enqueue?

Reply:
Yes, not that this is technically wrong :P